requester/requester.go

```diff
@@ -83,36 +92,72 @@ func (b *Work) Run() {
 	start := time.Now()
 	c := make(chan os.Signal, 1)
 	signal.Notify(c, os.Interrupt)
-
 	go func() {
 		<-c
 		// TODO(jbd): Progress bar should not be finalized.
-		newReport(b.N, b.results, b.Output, time.Now().Sub(start)).finalize()
+		newReport(b.N, b.results, b.Output, time.Now().Sub(start), b.EnableTrace).finalize()
 		os.Exit(1)
 	}()
 
 	b.runWorkers()
-	newReport(b.N, b.results, b.Output, time.Now().Sub(start)).finalize()
+	newReport(b.N, b.results, b.Output, time.Now().Sub(start), b.EnableTrace).finalize()
 	close(b.results)
 }
 
 func (b *Work) makeRequest(c *http.Client) {
 	s := time.Now()
 	var size int64
 	var code int
-
-	resp, err := c.Do(cloneRequest(b.Request, b.RequestBody))
+	var dnsDuration, connDuration, resDuration, reqDuration, delayTime time.Duration
+	req := cloneRequest(b.Request, b.RequestBody)
+	if b.EnableTrace {
+		trace := &httptrace.ClientTrace{
+			DNSStart: func(info httptrace.DNSStartInfo) {
+				dnsDuration = time.Now().Sub(s)
```

Comment:
reusing the same var for start is confuses the reader.

var dnsStart, connStart, ....
var dnsDuration, connDuration, resDuration, reqDuration, delayTime time.Duration